code/modules/mob/living/carbon/human/human_stripping.dm

```diff
@@ -129,13 +129,9 @@ GLOBAL_LIST_INIT(strippable_human_items, create_strippable_list(list(
 		return
 
 	var/obj/item/clothing/under/uniform = sourcemob.w_uniform
-	if(!LAZYLEN(uniform.accessories))
-		return FALSE
-	var/obj/item/clothing/accessory/accessory = LAZYACCESS(uniform.accessories, 1)
-	if(LAZYLEN(uniform.accessories) > 1)
-		accessory = tgui_input_list(user, "Select an accessory to remove from [uniform]", "Remove accessory", uniform.accessories)
-	if(!istype(accessory))
-		return
+
+	var/obj/item/clothing/accessory/accessory = uniform.pick_accessory_to_remove(user, sourcemob)
```

Comment:
It is not necessarily guaranteed `accessory` will be set to anything here (normally the UI won't offer this button if there isn't one - but better to have the handling), so you need to return here if nothing gets picked.